ERROR - 2024-12-29 14:17:59 --> Error connecting to the database: mysqli_sql_exception: Access denied for user ''@'localhost' (using password: NO) in C:\xampp\htdocs\Gestion_Exams\system\Database\MySQLi\Connection.php:186
Stack trace:
#0 C:\xampp\htdocs\Gestion_Exams\system\Database\MySQLi\Connection.php(186): mysqli->real_connect('localhost', '', Object(SensitiveParameterValue), '', 3306, '', 0)
#1 C:\xampp\htdocs\Gestion_Exams\system\Database\BaseConnection.php(421): CodeIgniter\Database\MySQLi\Connection->connect(false)
#2 C:\xampp\htdocs\Gestion_Exams\system\Database\BaseConnection.php(616): CodeIgniter\Database\BaseConnection->initialize()
#3 C:\xampp\htdocs\Gestion_Exams\system\Database\BaseBuilder.php(1644): CodeIgniter\Database\BaseConnection->query('SELECT 1\nFROM `...', Array, false)
#4 C:\xampp\htdocs\Gestion_Exams\system\Validation\StrictRules\Rules.php(232): CodeIgniter\Database\BaseBuilder->get()
#5 C:\xampp\htdocs\Gestion_Exams\system\Validation\Validation.php(339): CodeIgniter\Validation\StrictRules\Rules->is_unique('hatim@gmail.com', 'email', Array, NULL, 'email')
#6 C:\xampp\htdocs\Gestion_Exams\system\Validation\Validation.php(204): CodeIgniter\Validation\Validation->processRules('email', 'email', 'hatim@gmail.com', Array, Array, 'email')
#7 C:\xampp\htdocs\Gestion_Exams\system\Controller.php(137): CodeIgniter\Validation\Validation->run()
#8 C:\xampp\htdocs\Gestion_Exams\app\Controllers\registerController.php(26): CodeIgniter\Controller->validate(Array)
#9 C:\xampp\htdocs\Gestion_Exams\system\CodeIgniter.php(933): App\Controllers\RegisterController->attemptRegister()
#10 C:\xampp\htdocs\Gestion_Exams\system\CodeIgniter.php(509): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\RegisterController))
#11 C:\xampp\htdocs\Gestion_Exams\system\CodeIgniter.php(355): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#12 C:\xampp\htdocs\Gestion_Exams\system\Boot.php(325): CodeIgniter\CodeIgniter->run()
#13 C:\xampp\htdocs\Gestion_Exams\system\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#14 C:\xampp\htdocs\Gestion_Exams\public\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#15 {main}

Next CodeIgniter\Database\Exceptions\DatabaseException: Access denied for user ''@'localhost' (using password: NO) in C:\xampp\htdocs\Gestion_Exams\system\Database\MySQLi\Connection.php:231
Stack trace:
#0 C:\xampp\htdocs\Gestion_Exams\system\Database\BaseConnection.php(421): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 C:\xampp\htdocs\Gestion_Exams\system\Database\BaseConnection.php(616): CodeIgniter\Database\BaseConnection->initialize()
#2 C:\xampp\htdocs\Gestion_Exams\system\Database\BaseBuilder.php(1644): CodeIgniter\Database\BaseConnection->query('SELECT 1\nFROM `...', Array, false)
#3 C:\xampp\htdocs\Gestion_Exams\system\Validation\StrictRules\Rules.php(232): CodeIgniter\Database\BaseBuilder->get()
#4 C:\xampp\htdocs\Gestion_Exams\system\Validation\Validation.php(339): CodeIgniter\Validation\StrictRules\Rules->is_unique('hatim@gmail.com', 'email', Array, NULL, 'email')
#5 C:\xampp\htdocs\Gestion_Exams\system\Validation\Validation.php(204): CodeIgniter\Validation\Validation->processRules('email', 'email', 'hatim@gmail.com', Array, Array, 'email')
#6 C:\xampp\htdocs\Gestion_Exams\system\Controller.php(137): CodeIgniter\Validation\Validation->run()
#7 C:\xampp\htdocs\Gestion_Exams\app\Controllers\registerController.php(26): CodeIgniter\Controller->validate(Array)
#8 C:\xampp\htdocs\Gestion_Exams\system\CodeIgniter.php(933): App\Controllers\RegisterController->attemptRegister()
#9 C:\xampp\htdocs\Gestion_Exams\system\CodeIgniter.php(509): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\RegisterController))
#10 C:\xampp\htdocs\Gestion_Exams\system\CodeIgniter.php(355): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\Gestion_Exams\system\Boot.php(325): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\Gestion_Exams\system\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
#13 C:\xampp\htdocs\Gestion_Exams\public\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
#14 {main}
CRITICAL - 2024-12-29 14:17:59 --> CodeIgniter\Database\Exceptions\DatabaseException: Unable to connect to the database.
Main connection [MySQLi]: Access denied for user ''@'localhost' (using password: NO)
[Method: POST, Route: register]
in SYSTEMPATH\Database\BaseConnection.php on line 467.
 1 SYSTEMPATH\Database\BaseConnection.php(616): CodeIgniter\Database\BaseConnection->initialize()
 2 SYSTEMPATH\Database\BaseBuilder.php(1644): CodeIgniter\Database\BaseConnection->query('SELECT 1
FROM `users`
WHERE `email` = :email:
 LIMIT 1', [...], false)
 3 SYSTEMPATH\Validation\StrictRules\Rules.php(232): CodeIgniter\Database\BaseBuilder->get()
 4 SYSTEMPATH\Validation\Validation.php(339): CodeIgniter\Validation\StrictRules\Rules->is_unique('hatim@gmail.com', 'email', [...], null, 'email')
 5 SYSTEMPATH\Validation\Validation.php(204): CodeIgniter\Validation\Validation->processRules('email', 'email', 'hatim@gmail.com', [...], [...], 'email')
 6 SYSTEMPATH\Controller.php(137): CodeIgniter\Validation\Validation->run()
 7 APPPATH\Controllers\registerController.php(26): CodeIgniter\Controller->validate([...])
 8 SYSTEMPATH\CodeIgniter.php(933): App\Controllers\RegisterController->attemptRegister()
 9 SYSTEMPATH\CodeIgniter.php(509): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\RegisterController))
10 SYSTEMPATH\CodeIgniter.php(355): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 SYSTEMPATH\Boot.php(325): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Boot.php(67): CodeIgniter\Boot::runCodeIgniter(Object(CodeIgniter\CodeIgniter))
13 FCPATH\index.php(56): CodeIgniter\Boot::bootWeb(Object(Config\Paths))
